volatility3/framework/plugins/linux/mountinfo.py

```diff
@@ -0,0 +1,227 @@
+            mnt_type = self._get_symbol_fullname("vfsmount")
+
+        for mount in mnt_namespace.list.to_list(mnt_type, "mnt_list"):
+            yield mount
+            
+    def _get_tasks_mountpoints(self, pids: Iterable[int]):
+        self._vmlinux = self.context.modules[self.config['kernel']]
+               
+        pid_filter = pslist.PsList.create_pid_filter(pids)
+        tasks = pslist.PsList.list_tasks(self.context, self.config['kernel'], filter_func=pid_filter)
+        
+        seen_namespaces = set()
+        for task in tasks:
+            if not (task and task.fs and task.fs.root and task.nsproxy and task.nsproxy.mnt_ns):
+                # This task doesn't have all the information required
+                continue
+
+            mnt_namespace = task.nsproxy.mnt_ns
+            mount_ns_id = mnt_namespace.get_inode()
+            
+            if self._show_mountpoints_per_namespace:
+                if mount_ns_id in seen_namespaces:
+                    continue
+                else:
+                    seen_namespaces.add(mount_ns_id)
+            
+            for mount in self._get_mnt_namespace_mountpoints(mnt_namespace):
+                yield task, mount, mount_ns_id
+
+    def _generator(self):
+        pids = self.config.get('pids')
+
+        for task, mnt, mnt_ns_id in self._get_tasks_mountpoints(pids=pids):
+            mnt_info = self.get_mountinfo(mnt, task)
+            if mnt_info is None:
+                continue
+
+            if self.config.get('mount-format'):
+                all_opts = set()
+                all_opts.update(mnt_info.mnt_opts)
+                all_opts.update(mnt_info.sb_opts)
+                all_opts_str = ",".join(all_opts)
+
+                extra_fields_values = [mnt_info.devname, mnt_info.path_root, mnt_info.mnt_type, all_opts_str]    
+            else:
+                mnt_opts_str = ",".join(mnt_info.mnt_opts)
+                fields_str = " ".join(mnt_info.fields)
+                sb_opts_str = ",".join(mnt_info.sb_opts)
+
+                extra_fields_values = [mnt_info.mnt_id, mnt_info.parent_id, mnt_info.st_dev, mnt_info.mnt_root_path, 
+                                       mnt_info.path_root, mnt_opts_str, fields_str,  mnt_info.mnt_type, 
+                                       mnt_info.devname, sb_opts_str]
+            
+            fields_values = [mnt_ns_id]
+            if not self._show_mountpoints_per_namespace:
+                fields_values.append(task.pid)
+            fields_values.extend(extra_fields_values)
+            
+            yield (0, fields_values)
+
+    def run(self):
+        if self.config.get('all-processes') and self.config.get('pids'):
+            raise ValueError("Unable to use --all-processes and specified a pid")
+        
+        # When no arguments are specified, it displays the mountpoints per namespace
+        self._show_mountpoints_per_namespace = not any([self.config.get('pids'), self.config.get('all-processes')])
```

Comment:
So it feels like there's 3 states here:
* pids set
* pids not set and all-processes
* pids not set and not all-processes

It might be helpful to consider if it's neccessary for all of them (I think basically, I'm not quite clear on what `show_mountpoints_per_namespace` would do and why it *wouldn't* output pids?

Reply:
Yes, it could be a bit confusing. In all the cases the code walks the whole list of tasks. The difference is in the output.
The --pids argument is just a PID filter which is used to call the pslist's PID filter. If used, it only shows the information about these specific PIDs. Later, I found a `namespace filter` instead (or on top) of the pid filter could be also useful. Thoughts?

On the other hand, the default behaviour is to group the information by `mount namespace` and avoid showing PIDs. That's in essence what "mount" does showing only /proc/**self**/mount instead of each process mount namespace. Basically, this default behaviour is to avoid repeating rows, showing a less verbose output. 
In general many processes share a given namespace, so otherwise it will show a large number of rows with basically the same information, only changing the PID column values. Here's where **--all-processes** come into play. Probably not the best argument name but with the later, instead of showing the mount points per namespace, it shows the mount points per namespace and PID. I hope that was helpful

Reply:
Sooooo... how does it know which pid to use if `pids` isn't set and `all-processes` isn't set?  Is that just the kernel namespace?  I think adding the namespace filter and clarifying when/how these filters act/what their defaults are, would be the best way out of this.  Something about `--all-processess` and the `--pids` flag doesn't sit right with me...  I don't think `--all-processes` is accurate enough for what it's doing.

Reply:
I ended up removing `--all-process` and adding a mount namespace filter `--mntns`. The PID column is only shown when a pid filter `--pids` is used. Otherwise, the output is per mount namespace, so the PID column doesn't make sense in the default behavior.

Reply:
Oh ok, I checked it. As per the [code](https://github.com/volatilityfoundation/volatility3/blob/3c11099e6621027777ac5a9eeb3f8fbdc5b39b4b/volatility3/framework/__init__.py#L50), the major only has to be the same. The minor module version requirement could be same or lower and the patch level is not checked at all.
hm, I think it would be good idea to use something like a standard python module [packaging.version](https://packaging.pypa.io/en/latest/version.html) for comparing versions instead of doing it manually.

Also, note that the exception message is not accurate. In the hypothetical case that the framework has a lower version than the module requirement, the message would be, for instance:
'Framework interface version 2 is an older revision than the required version 2.2'
instead of:
'Framework interface version 2.1 is an older revision than the required version 2.2'

```diff
--- /tmp/qRIePE___init__.py
+++ /home/gmoreira/vol3-latest/volatility3/framework/__init__.py
@@ -51,7 +51,7 @@
             if args[1] > interface_version()[1]:
                 raise RuntimeError(
                     "Framework interface version {} is an older revision than the required version {}".format(
-                        ".".join([str(x) for x in interface_version()[0:1]]), ".".join([str(x) for x in args[0:2]])))
+                        ".".join([str(x) for x in interface_version()[0:2]]), ".".join([str(x) for x in args[0:2]])))
 
 
 class NonInheritable(object):
```

Reply:
Still not certain I'm in favour of the plugin making the decision over what should be displayed.  I think I'd prefer that all the data was output and the UI (or program consuming the data) got to decide what to display, but if you really think it affect readers of the output far more than it would affect programs which may want consistent output, then I'm not gonna make it a showstopper...